cmd/cli/kubectl-kyverno/processor/policy_processor.go

```diff
@@ -312,15 +312,18 @@ func (p *PolicyProcessor) ApplyPoliciesOnResource() ([]engineapi.EngineResponse,
 				return nil, fmt.Errorf("failed to apply validating policies on resource %s (%w)", resource.GetName(), err)
 			}
 			for _, r := range reps.Policies {
-				response := engineapi.EngineResponse{
-					Resource: *reps.Resource,
-					PolicyResponse: engineapi.PolicyResponse{
-						Rules: r.Rules,
-					},
+				if len(r.Rules) > 0 {
+					response := engineapi.EngineResponse{
+						Resource: *reps.Resource,
+						PolicyResponse: engineapi.PolicyResponse{
+							Rules: r.Rules,
+						},
+					}
```

Comment:
I suggest we first check if the `rules` is empty. If it is, we should skip creating a response. For example:
```
if len(r.Rules) == 0 {
   continue
}
```